dbms/src/Storages/StorageFile.cpp

```diff
@@ -18,12 +18,20 @@
 
 #include <Common/escapeForFileName.h>
 #include <Common/typeid_cast.h>
+#include <Common/parseGlobs.h>
 
 #include <fcntl.h>
 
 #include <Poco/Path.h>
 #include <Poco/File.h>
 
+#include <re2/re2.h>
+#include <re2/stringpiece.h>
+#include <boost/filesystem.hpp>
```

Comment:
Can we use std::filesystem from C++17?

Reply:
There is no easy way to do it, but it is not very sad because std::filesystem is directly based on boost one.

Reply:
Why not?

Reply:
As far as I remember the version of filesystem library fixed because of internal repo. So std::filesystem cannot be compiled, only experimental/filesystem, which has another syntax.

Reply:
We use std::filesystem everywhere.

Reply:
grep of the code says the opposite - only boost one out of contrib....

Reply:
Now we can.

Reply:
No, we don't use boost::filesystem at all:

```
milovidov@milovidov-Latitude-E7470:~/work/ClickHouse/dbms$ find . -name '*.cpp' -or -name '*.h' | xargs grep 'boost/filesystem'
milovidov@milovidov-Latitude-E7470:~/work/ClickHouse/dbms$ find . -name '*.cpp' -or -name '*.h' | xargs grep '<filesystem>'
./src/IO/tests/write_buffer_aio.cpp:#include <filesystem>
./src/IO/tests/read_buffer_aio.cpp:#include <filesystem>
./src/Functions/addressToLine.cpp:#include <filesystem>
./src/Functions/filesystem.cpp:#include <filesystem>
./src/Interpreters/tests/users.cpp:#include <filesystem>
./src/Common/Exception.cpp:#include <filesystem>
./src/Common/SymbolIndex.cpp:#include <filesystem>
./src/Common/StackTrace.cpp:#include <filesystem>
./src/Common/tests/compact_array.cpp:#include <filesystem>
./src/Common/getExecutablePath.cpp:#include <filesystem>
./src/Storages/MergeTree/DiskSpaceMonitor.h:#include <filesystem>
./src/Storages/StorageDistributed.cpp:#include <filesystem>
./programs/performance-test/PerformanceTestSuite.cpp:#include <filesystem>
./programs/performance-test/PerformanceTest.cpp:#include <filesystem>
./programs/performance-test/PerformanceTestInfo.cpp:#include <filesystem>
```

Reply:
19 days ago, I am too slow, I see https://github.com/yandex/ClickHouse/commit/f3921ce3e0e4a141b7c27e7488698b19383fb397

Reply:
Good reason for starting to read all commits in master